### ISSUE TITLE
Potential higher compatibility option in command with TAB plugin; nametags

### ISSUE THREAD

**Issue by TheJoshue** [libraryaddict/LibsDisguises#742]
So, basically, some option like `usingTAB` with boolean value, where the default value (if `usingTAB` option is not present in command) is `false`, assuming you must enter this option and set it to true to actually confirm you want TAB plugin to handle nametags

How will this work:
`/disguise player "SomeRandomUsername123" setSkin Notch usingTAB true`

So now when you execute this command, if the `usingTAB` option is present and functional in plugin, it will let TAB plugin control nametags, fully, even for handling when player is entering a vehicles (mentioning this because i recently reported a bug related to vehicles that TAB plugin similarly/exactly fixes to what i suggested in https://github.com/libraryaddict/LibsDisguises/issues/741)

Basically - nametag compatibility with TAB, where TAB plugin will have priority in nametags if this option is present and set to true.

**Comment by TheJoshue** [comment]
@libraryaddict any chance you check this out?

**Comment by MotikaCraft-TheJoshua** [comment]
Sorry for bothering @libraryaddict, any chance?

**Comment by libraryaddict** [comment]
Almost marks the third time I forgot to respond.

Anyways is this a global server setting you're looking for? Or did you specifically need it to only work on some disguises but others have normal name tags.

I'm mainly wondering if your use case is already handled by the scoreboard settings, such as changing it to vanilla. Or maybe the armorstand option isn't selected on your server.

I don't think there's a way to control that individually though, which is why I asked the first question. I need to verify if there's a reason that it's not usable on an individual level already, but I believe it's likely because it probably either provides unnecessary overhead or it conflicts on some level with each other. Basically not so performant.

Though a setting that simply controls if nametags are handled or not is simpler yeah, it's also not something I'm too keen on as it's a bad workaround that wants to stick around eternally because it's now being used actively.

**Comment by libraryaddict** [comment]
Sorry I had some trouble reading that, I'm a little tired though. But one thing sticks out, set upside down actually relies specifically on being able to change your nametag to a certain mojang player name. Only when your player disguise has that exact name will it be upside down. To pretend it's not that name, the name is hidden and armorstand is used to show the name on the player disguise instead. This is how it works for all disguises. 

I don't recall how tab changes names, but it is almost certain that it would need to hook into libs disguises, or libs disguises to hook into tab. Otherwise tab just doesn't know what name should be given.

Remember, if tab is looking at the real entity then it would never work on anything but actual players, and only for their own names.
If it works on packets, then it probably has no idea that dinnerbone is actually the player Jorge. I don't know how tab works, but basically it's not a simple answer for tab.

Off the top of my head, I know it uses armorstand. I'm guessing it also modifies scoreboard, and names are not calculated on the packet level.

Libs disguises needs to assign scoreboards in order to perform some functions, and this may be unexpected for tab. Or be exactly what tab needs to work but conflicts. Entirely depending on tab to handle it may result in bugs. 

So anyways, you're looking to only keep the original name of the player yeah? So you don't expect to have the tab name of other players such as offline. I don't remember exactly but I think tab doesn't handle names as well when you're disguised as it caused two sets of names to appear. Could be an option in its settings and you just need to stop using libs disguises to set names. Or set the name in libs disguises to empty to hide it.

**Comment by MotikaCraft-TheJoshua** [comment]
TAB v4 used armorstands only when unlimited-nametags option is enabled
It also supported unlimited multiple lines at the same time

However that feature was removed with TAB v5, where it purely relies on native vanilla nametag, only 2 lines (where that lower one is very limited to only value and fixed text...)
However to fix at least for 1.19.4+ clients you can use UnlimitedNametags plugin which for all 1.19.4+ viewers overrides regular TAB nametag and replaces with custom display entities text display entity or whatever it is ..... they work together very well, not double showing anything; and yes i know that since 1.20.3+ (i think) belowname objective can be almost fully customized, it's a good thing too but still limited with number of lines (i am only saying this for anyone who reads this if they need this info and why this is useful)

To not go far away from the topic,

> So anyways, you're looking to only keep the original name of the player yeah?

yes, option to not hide or not modify initial nametag by LibsDisguises even if gameprofile of player is set to Dinnerbone (upsidedown) or however it is handled, i am not very technical so i cannot really explain some things with 100% accuracy as i don't know how it actually works, but i assume that when you set player upside down he has that gameprofile of upside down player which is Dinnerbone/Grumm and by that their nametags apply above player head automatically(?) and that's why we cannot override it(?)

Potentially to not even hide nametag visibility as it still needs to be visible for TAB plugin to modify it in their configured way but also not to show that text which you define at `/disguiseplayer TheJoshua player "<HERE>"`
This could allow us for 2 native lines of nametag shown with vanilla, controlled by TAB (in this case)

**Comment by libraryaddict** [comment]
Again, have you tested this with vanilla setting in libs disguises configuration

**Comment by libraryaddict** [comment]
Libs disguises in scoreboard setting, should be displays.yml
Nametags has a comment for vanilla, it's all upper case name

**Comment by MotikaCraft-TheJoshua** [comment]
I have set it to VANILLA now and reloaded plugin

Plugin version: `LibsDisguises version 11.0.5` (i am not sure which dev build but after this i'll download latest)

command used:
`/disguiseplayer TheJoshua player "&4✪&r &8[&4&oAdmin&8] &4TheJoshua&r" setSkin TheJoshua setUpsideDown`

viewer (1.8 client on 1.21.1 server): 
![Image](https://github.com/user-attachments/assets/464012b4-5bc2-4e67-a529-055bd49ed838)

Note:
Is something changed here? because i didn't know it functions like this, it feels very off;
What i mean is that it spawns NPC which follows you... i feel this was already a thing before(?)
https://i.imgur.com/Pdylhsl.gif

**Comment by MotikaCraft-TheJoshua** [comment]
Sorry, i've just read i should restart server when changing this option you told me to test with VANILLA
After restarting, exactly same behaviour

btw, is this NPC spawning togglable?

**Comment by MotikaCraft-TheJoshua** [comment]
i think this should be like an attribute setting or whatever it is called
read this command example:

`/disguiseplayer TheJoshua player "&4✪&r &8[&4&oAdmin&8] &4TheJoshua&r" setSkin TheJoshua setUpsideDown setTABNametagOverride`

if possible on individual-level (per-player), `setTABNametagOverride` will make sure that (eligible; if permission-based disguise viewing gets implemented https://github.com/libraryaddict/LibsDisguises/issues/801) viewers will not see LibsDisguises set nametag but TAB's one (LibsDisguise won't even bother with "nametagging" player)

**Comment by libraryaddict** [comment]
Yes I need to look at what tab needs to work
